perceval/backends/core/weblate.py

```diff
@@ -0,0 +1,374 @@
+MIN_RATE_LIMIT = 10
+# Default sleep time and retries to deal with connection/server problems
+DEFAULT_SLEEP_TIME = 1
+MAX_RETRIES = 5
+
+CATEGORY_CHANGES = "changes"
+
+
+logger = logging.getLogger(__name__)
+
+
+class Weblate(Backend):
+    """Weblate backend for Perceval.
+
+    This class retrieves the changes for a given list of
+    keywords via the Google API. To initialize this class a list
+    of keywords is needed.
+
+    :param url: the URL of a Weblate instance
+    :param project: a Weblate project
+    :param api_token: Weblate API token
+    :param tag: label used to mark the data
+    :param archive: archive to store/retrieve items
+    :param max_retries: number of max retries to a data source
+        before raising a RetryError exception
+    :param sleep_for_rate: sleep until rate limit is reset
+    :param min_rate_to_sleep: minimum rate needed to sleep until
+         it will be reset
+    :param sleep_time: time (in seconds) to sleep in case
+        of connection problems
+    :param ssl_verify: enable/disable SSL verification
+    """
+    version = '0.1.0'
+
+    CATEGORIES = [CATEGORY_CHANGES]
+
+    def __init__(self, url, project=None, api_token=None, tag=None, archive=None,
+                 sleep_for_rate=False, min_rate_to_sleep=MIN_RATE_LIMIT,
+                 max_retries=MAX_RETRIES, sleep_time=DEFAULT_SLEEP_TIME, ssl_verify=True):
+
+        super().__init__(url, tag=tag, archive=archive, ssl_verify=ssl_verify)
+
+        self.api_token = api_token
+        self.project = project
+        self.max_retries = max_retries
+        self.sleep_time = sleep_time
+        self.sleep_for_rate = sleep_for_rate
+        self.min_rate_to_sleep = min_rate_to_sleep
+
+        self.client = None
+
+    def search_fields(self, item):
+        """Add search fields to an item.
+
+        It adds the values of `metadata_id` plus the `project`.
+
+        :param item: the item to extract the search fields values
+
+        :returns: a dict of search fields
+        """
+        search_fields = {
+            DEFAULT_SEARCH_FIELD: self.metadata_id(item)
+        }
+
+        return search_fields
+
+    def fetch(self, category=CATEGORY_CHANGES, from_date=DEFAULT_DATETIME):
+        """Fetch changes from Weblate API.
+
+        The method retrieves the changes that occurred on a project via the Weblate API.
+
+        :param category: the category of items to fetch
+
+        :returns: a generator of data
+        """
+        if not from_date:
+            from_date = DEFAULT_DATETIME
+
+        kwargs = {
+            'from_date': from_date
+        }
+        items = super().fetch(category, **kwargs)
+
+        return items
+
+    def fetch_items(self, category, **kwargs):
+        """Fetch Change items
+
+        :param category: the category of items to fetch
+        :param kwargs: backend arguments
+
+        :returns: a generator of items
+        """
+        logger.info("Fetching data")
```

Comment:
Can we improve this message with something like:
`logger.info("Fetching %s items on '%s' from '%s", category, self.url, from_date)`. You will need to put this under the `from_date` variable.